File: 1010/code2.c
#include <stdio.h>
#include <stdlib.h>

long long comb(int, int);

int main(){
    int tc = 0;
    int n, m, T = 0;
    scanf("%d", &tc);

    for(T=0; T<tc; T++) {
        scanf("%d", &n);
        scanf("%d", &m);
        printf("%lld\n", comb(m,n));
    }
    return 0;
}


long long comb(int n, int r){
    long long res = 1;
    long long tmp = 1;
    int i = 0;

    if(n/2 < r)
        r = n - r;
    
    for(i=n; i>= n-(r-1); i--)
        res *= i;
    for(i=r; i > 0; i--)
        tmp *= i;
    printf("%lld %lld\n", res, tmp);
    return res/tmp;
}

File: 1010/code2.js

var inp = require('fs').readFileSync('input').toString().split('\n')
var tc = inp[0] - 0
var res = ""

for (var T = 1; T <= tc; T++) {
    var tmp = inp[T].split(' ')
    var n = tmp[0] - 0
    var m = tmp[1] - 0

    if(n == m){ // 연산량을 줄임!
        res += 1 + "\n"
    } else {
        res += comb(m,n) + "\n"
    }

}
console.log(res)


function comb(n, r){
    var res = 1
    var tmp = 1

    // 이걸 안해주면, 29C27 같이 큰 수가 될 때 오차가 생긴다!
    if(n/2 < r)
        r = n - r

    for(var i=n; i>= n-(r-1); i--)
        res *= i
    for(var i=r; i > 0; i--)
        tmp *= i
    return res/tmp
}